components/sup/src/manager/service/config.rs

```diff
@@ -227,6 +234,18 @@ impl ServiceConfig {
         self.needs_write = false;
         Ok(should_restart)
     }
+
+    pub fn reload_package(&mut self, package: &PackageInstall) -> Result<()> {
+        self.config_root = match self.config_root_from_config {
```

Comment:
Same comment here [as above](https://github.com/habitat-sh/habitat/pull/1776/files#r101658477). This form causes an additional allocation